### PR TITLE
ci: Clean-up elastic network interface

### PR DESCRIPTION
**Pull request by engedaam**
Fixes #N/A 

**Description**
- Our E2E testing is seeing an increased number of ENI leaks. Adding ENIs are part of our clean-up script to delete any leaks.
 
**How was this change tested?**
- `/karpenter` snapshot

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

File: .github/actions/e2e/cleanup/action.yaml [hunk 1]
```diff
@@ -42,6 +42,22 @@ runs:
...
           fi
         done
+    - name: delete-cluster
+      shell: bash
+      run: |
+        eksctl delete cluster --name ${{ inputs.cluster_name }} --timeout 60m --wait || true
+     - name: delete-network-interfaces
+      shell: bash
+      run: |
+        aws ec2 describe-network-interfaces \
```
Comment (jonathan-innis) [inline .github/actions/e2e/cleanup/action.yaml:52]: Sanity check: does this work correctly if the output of this function is empty?

**Review by jonathan-innis — APPROVED** [review]

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

File: .github/actions/e2e/cleanup/action.yaml (hunk 1 above)
Comment (engedaam) [reply .github/actions/e2e/cleanup/action.yaml:52]: Confirmed, works as expected if the function is an empty string

**Review by bwagner5 — APPROVED** [review]
lgtm

File: test/hack/cleanup/main.go [hunk 2]
```diff
@@ -441,6 +443,71 @@ func (ip *instanceProfile) Cleanup(ctx context.Context, names []string) ([]strin
...
+}
+
+func (e *eni) Type() string {
+	return "ElasticNetworkInterface"
+}
+
+func (e *eni) Get(ctx context.Context, expirationTime time.Time) (ids []string, err error) {
+	var nextToken *string
+	for {
+		out, err := e.ec2Client.DescribeNetworkInterfaces(ctx, &ec2.DescribeNetworkInterfacesInput{
```
Comment (bwagner5) [inline test/hack/cleanup/main.go:457]: this manual pagination works fine, but you could also use the built-in paginator: https://aws.github.io/aws-sdk-go-v2/docs/making-requests/#using-paginators

Comment (engedaam) [reply test/hack/cleanup/main.go:457]: I will update the paginator for all the resources in a separate PR

**Pull request merged**